cmake/prefix.cmake

```diff
@@ -20,9 +20,7 @@ if (NOT DEV_BUILD AND
     ((CMAKE_C_COMPILER_ID MATCHES "Clang" AND (CMAKE_C_COMPILER_VERSION VERSION_GREATER "3.8" OR CMAKE_C_COMPILER_VERSION VERSION_EQUAL "3.8")) OR
     (CMAKE_C_COMPILER_ID STREQUAL "GNU" AND (CMAKE_C_COMPILER_VERSION VERSION_GREATER "4.4.7" OR CMAKE_C_COMPILER_VERSION VERSION_EQUAL "4.4.7"))))
     # Clang integrated assembler does not support this option.
-    set(CMAKE_REQUIRED_FLAGS "-Wa,--debug-prefix-map")
-    check_c_source_compiles("int main(void) {}" HAVE_AS_DEBUG_PREFIX_MAP)
-    unset(CMAKE_REQUIRED_FLAGS)
+    check_c_compiler_flag(-Wa,--debug-prefix-map== HAVE_AS_DEBUG_PREFIX_MAP)
     if (HAVE_WA_DEBUG_PREFIX_MAP)
```

Comment:
Why do you check `HAVE_WA_DEBUG_PREFIX_MAP` while the macro above sets `HAVE_AS_DEBUG_PREFIX_MAP`?

Reply:
Looks like a made a typo: fixed it.